src/main/java/frc/robot/subsystems/BallPath/Shooter/PIDShooterImpl.java

```diff
@@ -68,6 +97,72 @@ public void setShotPosition(ShotPosition shotPosition) {
         this.requestedPosition = shotPosition;
     }
 
+    public double getSetpointHood(double distance){
+        double hoodDif, distDif, difFromUpper, percentToAdd, amountToAdd;
+        double returnAmount = 0;
+        Integer Fender = 100_000;
+        Integer Tarmac = 230_000;
+        Integer CloseLaunchPad = 300_000; // pls test
+        Integer FarLaunchPad = 330_000; // pls test
+        Integer HumanPlayer = 400_000; // pls test
+        double[] Distances = {55.0, 153.0, 202.95, 244.77, 305.66};
+        Integer[] HoodValues = {100_000, 230_000, 300_000, 330_000, 400_000};
+
+        // LinkedHashMap<Double, Integer> foundValues = new LinkedHashMap<>();
+        // foundValues.put(55.0, Fender);
+        // foundValues.put(153.0, Tarmac);
+        // foundValues.put(202.95, CloseLaunchPad);
+        // foundValues.put(244.77, FarLaunchPad);
+        // foundValues.put(305.66, HumanPlayer);
+        int i = 0;
+        for (Double key : Distances) {
+            if(distance < key){
+                distDif = Distances[i] - Distances[i-1];
```

Comment:
This looks like maybe it happens to work because of the `if` condition, but the `i-1` here is suspicious. On the first iteration you have `i == 0`, so you're trying to access `distances[-1]`. That's an out of bounds index in Java.

Reply:
Yes, but this shouldnt happen as we dont even see the limelight on the fender shot. The minimum value we get would be the auto shot or whatever it was called

Reply:
"this shouldn't happen" is not reassuring when this is a potential out of bounds exception that crashes the thread.

Reply:
please add some other kind of logical guard to ensure that the `distances[-1]` *cannot* happen, not that it "should not happen so long as the sensor doesn't pick up the wrong kind of static or something"